File: 프로그래머스/0/181928. 이어 붙인 수/이어 붙인 수.py
def solution(num_list):
    odd_numbers = []
    even_numbers = []
    
    for i in num_list:
        if i % 2 != 0:
            odd_numbers.append(str(i))
        else:
            even_numbers.append(str(i))
    
    return int(''.join(odd_numbers)) + int(''.join(even_numbers))

File: 프로그래머스/0/181928. 이어 붙인 수/이어 붙인 수.java
class Solution {
    public int solution(int[] num_list) {
        StringBuilder evenStr = new StringBuilder();
        StringBuilder oddStr = new StringBuilder();

        for(int num : num_list) {
            if(num % 2 == 0) {
                evenStr.append(num);
            } else {
                oddStr.append(num);
            }
        }
        
        int even = evenStr.length() > 0 ? Integer.parseInt(evenStr.toString()) : 0;
        int odd = oddStr.length() > 0 ? Integer.parseInt(oddStr.toString()) : 0;

        return even + odd;
    }
}
